### PR TITLE
feat: added total basket and total cost for p10x

### PR DESCRIPTION
**Pull request by PaulRouss3l**
Ticket : https://app.clickup.com/t/20427503/INT-473

### CODE REVIEW

File: cartridges/int_alma/cartridge/scripts/helpers/almaCheckoutHelper.js [hunk 1]
```diff
@@ -58,10 +58,14 @@ function getPaymentCoreInfo(plan) {
  */
 function getCreditInfo(plan, currencyCode) {
     var costOfCredit = formatCurrency(plan.customer_total_cost_amount / 100, currencyCode);
+    var purchaseAmount = formatCurrency(plan.purchaseAmount, currencyCode);
+    var totalCost = formatCurrency(plan.purchaseAmount + (plan.customer_total_cost_amount / 100), currencyCode);
```
Comment (syjust-alma) [inline cartridges/int_alma/cartridge/scripts/helpers/almaCheckoutHelper.js:62]: 1. `formatCurrency` is only available in `sfcc-cartridge-dev` repository... how does it works ? where `formatCurrency` come from here ? (`*/cartridge/scripts/util/formatting` not seem's to be a part of current repository)
2. I'am not sure that `formatMoney` or `Money` (used by `formatCurrency`) will round given amount. Can you confirm that for me please (or explain me why we don't use `Math.round()` here) ?

File: cartridges/int_alma/cartridge/scripts/helpers/almaPlanHelper.js [hunk 2]
```diff
@@ -135,6 +135,7 @@ function getPlansForCheckout(locale, currentBasket) {
     });
 
     plans = almaUtilsHelpers.map(plans, function (plan) {
+        plan.purchaseAmount = purchaseAmount; // eslint-disable-line no-param-reassign
```
Comment (syjust-alma) [inline cartridges/int_alma/cartridge/scripts/helpers/almaPlanHelper.js:138]: Are you sure that remove `Math.round()` bellow will not make problems here ?

**Review by syjust-alma — COMMENTED** [review]
Can you try to develop some tests (unit test or e2e) where amount is used without `Math.round()` to verify everything will works as attempted

File: cartridges/int_alma/cartridge/scripts/helpers/almaPlanHelper.js (hunk 2 above)
Comment (PaulRouss3l) [reply cartridges/int_alma/cartridge/scripts/helpers/almaPlanHelper.js:138]: yes :), as discussed together we should remove it here and in the other place too

**Comment by PaulRouss3l** [comment]
> Can you try to develop some tests (unit test or e2e) where amount is used without Math.round() to verify everything will works as attempted

yes but in another PR

File: cartridges/int_alma/cartridge/scripts/helpers/almaCheckoutHelper.js [hunk 3]
```diff
@@ -162,6 +166,10 @@ function isFragmentActivated() {
  * @returns {array} a plan Object understandable for Eligibility
  */
 function formatPlanForCheckout(plan, currencyCode) {
+    var logger = require('dw/system/Logger');
+    logger.error(plan.purchaseAmount.toString());
```
Comment (syjust-alma) [inline cartridges/int_alma/cartridge/scripts/helpers/almaCheckoutHelper.js:170]: why we have an `error` logged here ?

if log is required, please add some context, not just the `purchaseAmount` as string 😉

Comment (PaulRouss3l) [reply cartridges/int_alma/cartridge/scripts/helpers/almaCheckoutHelper.js:170]: because I've left my debug :( thanks for catching that ... )

**Review by syjust-alma — APPROVED** [review]

**Pull request merged**